lib/Plack/Request.pm

```diff
@@ -68,7 +68,15 @@ sub cookies {
         # trim leading trailing whitespace
         $pair =~ s/^\s+//; $pair =~ s/\s+$//;
 
-        my ($key, $value) = map URI::Escape::uri_unescape($_), split( "=", $pair, 2 );
+        my ($key, $value) = split( "=", $pair, 2 );
+
+        $key = URI::Escape::uri_unescape($key);
+
+        # Values can be quoted
+        $value =~ s/\A"//;
```

Comment:
i don't mean to be pedantic but should we only unquote if the quotes are balanced?